src/v/transform/rpc/client.cc

```diff
@@ -96,6 +97,46 @@ cluster::errc map_errc(std::error_code ec) {
     }
     return cluster::errc::timeout;
 }
+
+template<typename Func>
+std::invoke_result_t<Func> retry(Func func) {
+    int attempts = 0;
+    while (true) {
+        ++attempts;
+        auto fut = co_await ss::coroutine::as_future<
+          typename std::invoke_result_t<Func>::value_type>(func());
```

Comment:
probably better to use futurator / futurize_invoke? It works for non future returning functions as well.

Reply:
Probably? I've added new commits for this.